### ISSUE TITLE
Geolocation in location modal 🌍

### ISSUE THREAD

**Issue by richardolsson** [zetkin/app.zetkin.org#1179]
## Description
A user should be able to navigate the map to the location that they're in using geolocation on the device that they're using to interact with Zetkin.

## Relevant Job Stories
None

## Prerequisites
Don't start this until #1172 is done.

## Requirements
- [x] In the modal search field, there should be a pin icon button
- [x] Clicking the pin should request the user's location from the browser
- [x] Once the location is retrieved, the map should be panned and zoomed to the location of the user

## Possible implementations
None

## Design specifications
![image](https://user-images.githubusercontent.com/550212/228886235-9401d1dd-5d1c-44a1-8ae6-f2cd351b23db.png)

## Open questions
None

**Comment by richardolsson** [comment]
@dothelooping: This was completed by @river-bbc over the weekend in #1264 so you can move on to another issue. I hope you didn't get too far and feel like time was wasted.
